tools/generator/dfg/nrf/nrf_device_tree.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2013-2016, Niklas Hauser
+# Copyright (c)      2016, Fabian Greif
+# Copyright (c)      2020, Hannes Ellinger
+# All rights reserved.
+
+import math
+import logging
+
+from ..device_tree import DeviceTree
+from ..input.xml import XMLReader
+
+from .nrf_identifier import NRFIdentifier
+
+LOGGER = logging.getLogger('dfg.nrf.reader')
+
+class NRFDeviceTree:
+    """ NRFDeviceTree
+    This NRF specific part description file reader knows the structure and
+    translates the data into a platform independent format.
+    """
+
+    @staticmethod
+    def _properties_from_file(filename):
+        device_file = XMLReader(filename)
+        p = {}
+
+        LOGGER.info("Parsing Device File For '%s'", device_file.query("//device/name")[0])
+
+        partname = device_file.query("//device/name")[0].text
+
+        # dirty hack because af inconsistent part names in .svd files
+        if partname == 'nrf52':
+            partname = 'nrf52832'
+        elif partname == 'nrf51':
+            partname = 'nrf51822'
+
+        did = NRFIdentifier.from_string(partname.lower())
+        p['id'] = did
+
+        # information about the core and architecture
+        core = device_file.query("//device/cpu/name")[0].text.lower().replace("cm", "cortex-m")
+        if device_file.query("//device/cpu/fpuPresent")[0].text == '1':
+            core += "f"
```

Comment:
This doesn't seem to work right, the nrf52 files all have `cortex-m4`, instead of `cortex-m4f`.

Reply:
Huh, interesting. Looks like only the 32, 33 and 40 version of the chip feature an FPU:

https://infocenter.nordicsemi.com/index.jsp?topic=%2Fstruct_nrf52%2Fstruct%2Fnrf52.html

Reply:
Oh, haven't seen a M4 without FPU yet, let's see im modm generates the right code. (I believe all relevant code only check for the `f` or `fd` prefix, so it probably just works).